File: 258.add-digits.js
/*
 * @lc app=leetcode id=258 lang=javascript
 *
 * [258] Add Digits
 */

// @lc code=start
/**
 * @param {number} num
 * @return {number}
 */
var addDigits = function (num) {
    while (num > 9) {
        num = num.toString().split('').reduce((a, b) => (+a) + (+b));
    }
    return num;
};

let num = 38

// @lc code=end

File: 258.add-digits.py
#
# @lc app=leetcode id=258 lang=python
#
# [258] Add Digits
#

# @lc code=start
class Solution(object):
    def addDigits(self, num):
        """
        :type num: int
        :rtype: int
        """

        if num == 0:
            return 0

        return (num - 1) % 9 + 1

# @lc code=end
